src/features/board/boardSlice.ts

```diff
@@ -28,11 +29,14 @@ const BoardSlice = createSlice({
         attack: (state, action: PayloadAction<[number, number]>) => {
             state.playerSide.cards[action.payload[0]].health -= state.opponentSide.cards[action.payload[1]].attack
             state.opponentSide.cards[action.payload[1]].health -= state.playerSide.cards[action.payload[0]].attack
+        },
+        selectCard: (state, action: PayloadAction<number>) => {
+            state.selectedCard = state.playerSide.cards[action.payload];
```

Comment:
So here you could simply do : 
```javascript
state.selectedCard = action.payload;
```
So far I dont see why we would need the full card in selectedCard but if you think it would be usefull for some reason feel free to tell why.